connectors/filtering/validation.py

```diff
@@ -105,6 +147,12 @@ def __eq__(self, other):
 
         return self.state == other.state and self.errors == other.errors
 
+    def to_dict(self):
```

Comment:
as an improvement for later, you could also make that class act as a dict, so it can be directly used 

see https://realpython.com/inherit-python-dict/